node/modules/chain.go

```diff
@@ -59,7 +59,7 @@ func ChainBlockService(bs dtypes.ExposedBlockstore, rem dtypes.ChainBitswap) dty
 }
 
 func MessagePool(lc fx.Lifecycle, us stmgr.UpgradeSchedule, mpp messagepool.Provider, ds dtypes.MetadataDS, nn dtypes.NetworkName, j journal.Journal, protector dtypes.GCReferenceProtector) (*messagepool.MessagePool, error) {
-	mp, err := messagepool.New(mpp, ds, us, nn, j)
+	mp, err := messagepool.New(context.Background(), mpp, ds, us, nn, j)
```

Comment:
Use `helpers.LifecycleCtx(lc fx.Lifecycle, mctx helpers.MetricsCtx)` to get a 'real' context (or at least cast from mctx helpers.MetricsCtx for metrics)